File: Project/textnode.cpp
#include "textnode.h"

TextNode::TextNode(QString family, QString color, QString size)
{

    this->family = family;
    this->color = color;
    this->size = size;


}

TextNode::TextNode(QString family, QString color, QString size, bool bold, bool italic, TextNode *parent)
{

    this->family = family;
    this->color = color;
    this->size = size;
    this->bold = bold;
    this->italic = italic;
    this->parent = parent;


}


TextNode:: ~TextNode()
{

    for (int i = 0; i < this->children.length(); ++i)
    {
        delete children.at(i);
    }

}

void TextNode::setText(QString text)
{
    this->text = text;
}


QString TextNode::getText()
{
   return text;

}


void TextNode::setSize(QString size)
{

    this->size = size;

}

QString TextNode::getSize()
{
    return size;

}

bool TextNode::isBold()
{
    return bold;
}

void TextNode::setBold(bool bold)
{
    this->bold = bold;
}

bool TextNode::isItalic()
{
    return italic;
}

void TextNode::setItalic(bool italic)
{
    this->italic = italic;
}

QString TextNode::getColor()
{
    return color;
}

void TextNode::setColor(QString s)
{
    this->color = color;
}

QString TextNode::getFamily()
{
    return family;
}

void TextNode::setFamily(QString family)
{
    this->family = family;
}

QList<TextNode *> TextNode::getNode()
{
    return children;
}

void TextNode::setAsChildren(TextNode *node)
{
    children.append(node);
}

int TextNode::getId()
{
    return id;
}

void TextNode::setId(int id)
{
    this->id = id;
}

TextNode *TextNode::getParent()
{
    return parent;
}

void TextNode::setParent(TextNode *parent)
{
    this->parent = parent;
}

void TextNode::setLink(bool link)
{
    this->link = link;
}

bool TextNode::isLink()
{
    return link;
}

bool TextNode::isIcon()
{
    return icon;
}

void TextNode::setIcon(bool icon)
{
   this->icon = icon;
}

QString TextNode::getIconText()
{
    return iconText;
}

void TextNode::setIconText(QString s)
{
    this->iconText = s;
}

QString TextNode::getHyperLink()
{
    return hyperlink;
}

void TextNode::setHyperLink(QString hyperlink)
{
    this->hyperlink = hyperlink;
}

File: src/components/TextNode.java
package components;

import java.util.ArrayList;
import java.util.List;

public class TextNode {
	
	private String text;
	
	private boolean bold;
	
	private boolean italic;
	
	private boolean icon;
	
	private String iconText;
	
	private int size;
	
	private String color;
	
	private String family;
	
	private boolean link;
	
	private String hyperlink;
	
	private ArrayList<TextNode> children;
	private TextNode parent;
	private int id;
	
	
	public TextNode(String family, String color, int size) {
		setSize(size);
		setColor(color);
		setFamily(family);
		setBold(false);
		setItalic(false);
		children = null;
		parent = null;
	}
	
	public TextNode(String family, String color, int size, boolean bold, boolean italic, TextNode parent) {
		setSize(size);
		setColor(color);
		setFamily(family);
		setBold(bold);
		setItalic(italic);
		children = null;
		setParent(parent);
	}
	



	public String getText() {
		return text;
	}
	
	public void setSize(int size) {
		this.size = size;
	}
	
	public int getSize() {
		return size;
	}

	public void setText(String text) {
		this.text = text;
	}

	public boolean isBold() {
		return bold;
	}

	public void setBold(boolean bold) {
		this.bold = bold;
	}

	public boolean isItalic() {
		return italic;
	}

	public void setItalic(boolean italic) {
		this.italic = italic;
	}

	public String getColor() {
		return color;
	}

	public void setColor(String color) {
		this.color = color;
	}

	public String getFamily() {
		return family;
	}

	public void setFamily(String family) {
		this.family = family;
	}

	public ArrayList<TextNode> getNode() {
		return children;
	}

	public void setAsChildren(TextNode node) {
		if (children == null)
			children = new ArrayList<TextNode>();
		this.children.add(node);
		//System.out.println("szöveg: " + node.getId() + " csatlakoztatva " + getId() + "-hez");
	}

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public TextNode getParent() {
		return parent;
	}

	public void setParent(TextNode parent) {
		this.parent = parent;
	}

	public void setLink(boolean link) {
		// TODO Auto-generated method stub
		this.link = link;
		
	}
	
	public boolean isLink() {
		return link;
	}

	public boolean isIcon() {
		return icon;
	}



	public String getIconText() {
		return iconText;
	}

	public void setIconText(String iconText) {
		this.iconText = iconText;
	}

	public void setIcon(boolean icon2) {
		// TODO Auto-generated method stub
		icon = icon2;
		
	}

	public String getHyperlink() {
		return hyperlink;
	}

	public void setHyperlink(String hyperlink) {
		this.hyperlink = hyperlink;
	}

}
